weston-ivi-shell/src/ivi-controller.c

```diff
@@ -26,13 +26,16 @@
  * ivi-layout.c in weston.
  */
 
+#include "config.h"
+
 #include <stdlib.h>
 #include <string.h>
 
+#include <sys/mman.h>
```

Comment:
When I compile these patches in our environment, I am getting new compiler warnings:
warning: implicit declaration of function 'unlink' [-Wimplicit-function-declaration]
warning: implicit declaration of function 'posix_fallocate' [-Wimplicit-function-declaration]
warning: implicit declaration of function 'close' [-Wimplicit-function-declaration]

I think you missed to add some header files